File: android/app/src/test/java/com/quartier/quartier/mock_models/MockAuthManager.kt
package com.quartier.quartier.mock_models

import com.quartier.quartier.R
import com.quartier.quartier.auth.AuthManager
import com.quartier.quartier.auth.AuthResult

//Imitate the behavior of supabase authentication

class MockAuthManager : AuthManager {
    val accounts = mutableMapOf(
        Pair("email@gmail.com", "Password")
    )

    override suspend fun emailSignIn(
        eml: String,
        pwd: String
    ): AuthResult {
        return if(accounts[eml] == pwd) AuthResult.Success
        else AuthResult.Error(R.string.auth_invalid_credentials)
    }

    override suspend fun emailSignUp(
        name: String,
        eml: String,
        pwd: String
    ): AuthResult {
        return if(accounts[eml] != null) AuthResult.Error(R.string.auth_user_exists)
        else AuthResult.Success
    }
}

File: ios/QuartierTests/MockModels/MockAuthManager.swift
//
//  MockAuthManager.swift
//  Quartier
//
//  Created by Emile Turcotte on 2025-07-21.
//

import Auth
import Foundation
@testable import Quartier

//Mock the behavior of the auth manager for viewmodel tests, checked against the accounts dictionary

class MockAuthManager: AuthManager {
    let accounts = ["email@gmail.com": "Password"]
    
    func login(email:String, pwd: String) async throws {
        if(accounts[email] != pwd) {
            throw AuthError.api(message: "Invalid login credentials", errorCode: .invalidCredentials, underlyingData: Data(), underlyingResponse: HTTPURLResponse(url: URL(string: "https://example.com")!, statusCode: 401, httpVersion: nil, headerFields: nil)!)
        }
    }
    
    func signup(email:String, pwd: String, name: String) async throws {
        if(accounts[email] != nil) {
            throw AuthError.api(message: "User already exists", errorCode: .userAlreadyExists, underlyingData: Data(), underlyingResponse: HTTPURLResponse(url: URL(string: "https://example.com")!, statusCode: 401, httpVersion: nil, headerFields: nil)!)
        }
    }
    
    func signout() async throws {}
}
